### DISCUSSION TITLE
airodump-ng rate setup

### DISCUSSION THREAD

**Discussion by farishajdarpasic** [aircrack-ng/aircrack-ng#2574] [General]
Hello. So, my project is related to localization of a robot. I need to know at each timestamp, which APs are visible for robot's WiFi reciever. But this airodump-ng command is running continously. I was wondering, is there a posibility, to somehow set the rate of scanning the available APs? So if I set a rate of 10Hz, then every 0.1 seconds, I would have list of visible APs for specific robot position, and I could store those observations to a file, and proceed with my method for localization.

Unfortunately airodump-ng is running continously, so that is making me problem (and as far as I experimented, I would prefer to use airodump-ng instead of iwlist). So, is there any way to achieve this? This would be useful thing for my project. Also, if there is no direct way, could you please suggest me how to simulate this - to lets say call this command every 0.1 second (but since it is running contiously it is making me problem). Or could you please tell me any other command that you know, that maybe can be used for my project. 

Thank you

**Comment by Mister-X-** [comment]
There is no such thing as scanning rate, for any wireless tool.

You can set the hopping to 0.1 second, using the `-f` parameter. It will change channel every 0.1 second.

**Reply by farishajdarpasic** [reply]
Problem is I need to call this command in ROS node. ROS node also has its own rate, so I have to combine them now somehow. But if this command runs continously, how then will loop get executed in ROs Publisher if it get stuck with this command running continously?

2nd thing, is there any way, so somehow write outputs of those hoppings in different file or at least in same file, separated by some dashed lines or similar? Please let me know if your command have some option for that, since I need to have format, where each element has timestamp and list of available APs for that timestamp.

**Reply by Mister-X-** [reply]
1. What is a ROS node?

2. As long as you keep running airodump-ng, it will write to the same file. Just parse it over and over. If you need it in a custom format, you'll need to write your own tool. I can help if needed, for a fee.

**Reply by farishajdarpasic** [reply]
Like a python script. But how can I parse it, when I dont know which wifi data corresponds to which time. How can I distinguish which set of APs are for different timestamps?

**Reply by Mister-X-** [reply]
What you need is really specific and you can't do that with airodump-ng or other tools I can think of. You'll need a custom tool to parse frames.

**Reply by farishajdarpasic** [reply]
I just need some command that runs once. If I call it, it runs one time and thats it, without continuin. Like iwlist. But your command is better than iwlist, I wish your is running only one scan...

**Reply by Mister-X-** [reply]
there is `timeout` you can use, but you'll need to adjust the write interval (default is every 4 seconds)

**Reply by farishajdarpasic** [reply]
In airodump-ng docs I dont see that option, what timeout you mean excatly?

**Reply by Mister-X-** [reply]
https://www.man7.org/linux/man-pages/man1/timeout.1.html

**Reply by farishajdarpasic** [reply]
Thank you for your time. After I try in Lab, I will see and potentially ask something

**Comment by shashankyld** [comment]
I would like to run the airodump-ng command in loop and save the output in different files continously. But it doesnt seem to work even when i use timeout. Can you try to do the same in your machine and share your results? 
In my case, I can see the airodump-ng command continue to run even when there is timeout or when i use threads with timeoutflag.

**Reply by shashankyld** [reply]
Also I am assuming now that - airodump-ng is not particulary designed to read wifi fingerprint info for a particular interface. I am planning to try out ```bash https://github.com/Mister-X-/wig-ng```.
